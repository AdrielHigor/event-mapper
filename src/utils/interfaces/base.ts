
export interface IEventForm {
  name?: string;
  description?: string;
  startDateTime?: string;
  endDateTime?: string;
}

export interface IEventFormValidation {
  name?: boolean;
  description?: boolean;
  startDateTime?: boolean;
  endDateTime?: boolean;
}

export interface DynamicObject {
  [key: string]: string | boolean | undefined
}

export interface ILocationResponse {
  id: number,
  point: {
    coordinates: number[],
    crs: {
      properties: {
        type: string,
      },
      type: string,
    },
    type: string,
  }
}

export interface IEventResponse {
  id: number,
  name: string,
  description: string,
  location: ILocationResponse,
  startDateTime: string,
  endDateTime: string,
  createdAt: string,
  updatedAt: string,
  deletedAt: string,
}

export interface IEventPost {
  name: string,
  description: string,
  startDateTime?: string,
  endDateTime?: string,
  location: {
    lat: number,
    lng: number,
  }
}

export interface IMockModalContainer {
  children: React.ReactNode;
}