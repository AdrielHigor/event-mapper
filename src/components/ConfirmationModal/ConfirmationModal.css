.confirmationContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.modalHeader {
  display: flex;
  height: 80%;
}

.description {
  text-align: center;
}

.options {
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  width: 100%;
  margin: 15px 0px;
}

.confirmButton {
  border: 0;
  border-radius: 5px;
  width: 160px;
  height: 40px;
  background-color: #00FF00;
  font-weight: bold;
  font-size: 14px;
}

.confirmButton:hover {
  background-color: #02b102;
  cursor: pointer;
}

.cancelButton {
  border: 0;
  border-radius: 5px;
  width: 160px;
  height: 40px;
  background-color: #ff0000;
  font-weight: bold;
  font-size: 14px;
}

.cancelButton:hover {
  background-color: #b10202;
  cursor: pointer;
}

@media (max-width: 420px) {
  h2 {
    text-align: center;
  }
  
  .options {
    flex-direction: column;
    justify-content: space-evenly;
  }

  .confirmButton {
    margin-bottom: 10px;
  }
}