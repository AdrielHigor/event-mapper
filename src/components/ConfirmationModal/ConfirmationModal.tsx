import React from "react";
import ModalContainer from "../ModalContainer/ModalContainer";
import "./ConfirmationModal.css";

interface IConfirmationModal {
  modalIsOpen: boolean;
  closeModal: (
    event: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<Element>
  ) => void;
  customClassName?: string;
  title: string;
  description: string;
  confirmLabel: string;
  cancelLabel: string;
  handleConfirm: () => void;
  handleCancel: () => void;
}

const ConfirmationModal = ({
  modalIsOpen,
  closeModal,
  customClassName,
  title,
  description,
  confirmLabel,
  cancelLabel,
  handleConfirm,
  handleCancel,
}: IConfirmationModal) => (
  <ModalContainer
    modalIsOpen={modalIsOpen}
    closeModal={closeModal}
    customClassName={customClassName}
  >
    <div className="confirmation-container">
      <div className="modal-header">
        <h2>{title}</h2>
        <p className="description">{description}</p>
      </div>
      <div className="options">
        <button className="confirm-button" onClick={handleConfirm}>
          {confirmLabel}
        </button>
        <button className="cancel-button" onClick={handleCancel}>
          {cancelLabel}
        </button>
      </div>
    </div>
  </ModalContainer>
);

export default ConfirmationModal;
