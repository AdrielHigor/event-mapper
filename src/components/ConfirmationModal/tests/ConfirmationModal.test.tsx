import React from "react";
import renderer from "react-test-renderer";
import ConfirmationModal from "../ConfirmationModal";
import { IMockModalContainer } from "../../../utils/interfaces/base";

jest.mock("../ConfirmationModal.css", () => ({}));
jest.mock(
  "../../ModalContainer/ModalContainer",
  () =>
    ({ children }: IMockModalContainer) =>
      <div>{children}</div>
);

describe("ConfirmationModal", () => {
  let component: renderer.ReactTestRenderer;

  const defaultProps = {
    modalIsOpen: true,
    closeModal: jest.fn(),
    title: "Are you sure?",
    description: "This action cannot be undone.",
    confirmLabel: "Confirm",
    cancelLabel: "Cancel",
    handleConfirm: jest.fn(),
    handleCancel: jest.fn(),
  };

  beforeEach(() => {
    component = renderer.create(<ConfirmationModal {...defaultProps} />);
  });

  it("should render correctly", () => {
    const title = component.root.findByProps({
      className: "modal-header-title",
    });
    const description = component.root.findByProps({
      className: "description",
    });

    expect(title.props.children).toBe(defaultProps.title);
    expect(description.props.children).toBe(defaultProps.description);
    expect(component).toMatchSnapshot();
  });

  it("should call handleConfirm on confirm button click", () => {
    const confirmButton = component.root.findByProps({
      className: "confirm-button",
    });
    confirmButton.props.onClick();
    expect(defaultProps.handleConfirm).toHaveBeenCalledTimes(1);
  });

  it("should call handleCancel on cancel button click", () => {
    const cancelButton = component.root.findByProps({
      className: "cancel-button",
    });
    cancelButton.props.onClick();
    expect(defaultProps.handleCancel).toHaveBeenCalledTimes(1);
  });
});
