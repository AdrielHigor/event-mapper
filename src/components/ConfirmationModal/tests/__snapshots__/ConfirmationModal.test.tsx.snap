// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal should render correctly 1`] = `
<div>
  <div
    className="confirmation-container"
  >
    <div
      className="modal-header"
    >
      <h2
        className="modal-header-title"
      >
        Are you sure?
      </h2>
      <p
        className="description"
      >
        This action cannot be undone.
      </p>
    </div>
    <div
      className="options"
    >
      <button
        className="confirm-button"
        onClick={[MockFunction]}
      >
        Confirm
      </button>
      <button
        className="cancel-button"
        onClick={[MockFunction]}
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
